NO ISSUE: Merge Couchbase Server 8.0 changes into Capella

simon-dew opened the pull request:
This PR merges _most_ Couchbase Server 8.0 changes into the Capella branch.

It does not merge forward changes to do with the Search service or Vector Search / FTS Vector.

- The PRs which are not included here are [these](https://github.com/couchbaselabs/docs-devex/pulls?q=is%3Apr+label%3A8.0%2Cfts).
- The following non-PR changes are also not included here:
  - 382916f5329c667b909ae4c1a66082cbbb29de79
  - b005c293d41995eb20f976a5d28d13d60389f5d7
  - a38b065192598cd8fad3b68cfee608e5762b9d61
  - 67cdfb41e9b2f2a3cb97ec799c51298cadbbae5f
  - f89afa5dd7f28fd1e831f79b60870282e4a0433a
  - 72de1e817e0aae395fe484b779bcbe475ce06f5e

The changes in this PR need to be combined with AV-91970

> [!IMPORTANT]
> Rebase and merge! Do not squash and merge

RayOffiah committed:
[DOC-11988-7.6]: RBAC role change impact to eventing function
Minor corrections

simon-dew committed:
DOC-12443: Reorganize GSI documents (#261)

* Move pages to Indexes
* Move diagrams to Indexes
* Move examples to Indexes
* Move images to Indexes
* Move indexes concept doc back to docs-server
* Remove unused image files
* Update page aliases
* Update links to GSI pages
* Update coordinates of example files
* Update GSI image coordinates
* Update GSI diagram coordinates
* Driveby fix unsupported icons
* Minor changes to indexing overview and content order
* Move covering indexes, group agg pushdown
* Update links to covering indexes, grouping and agg pushdown
* Driveby fix broken link

simon-dew committed:
DOC-12312: streaming completed requests to local files (#273)

* Add Streaming Completed Requests
* Try to make tabs sync with only partial success
* Add system:completed_requests_history to sysinfo
* Add links to Admin REST API
* Update table formatting for sysinfo
* Add links to gzip and jq

simon-dew committed:
DOC-12652: Convert other server REST API references to use OpenAPI generator (#280)

* Update links to Query Settings REST API
* Update includes of Query Settings REST API
* Driveby fixes for CURL() function
* Update links to Query Settings examples
* Add table-style partial
* Add table-style for CURL() reference

RayOffiah committed:
[DOC-12901]: erroneous vector search text (#312)
Fix grammar in the pre-filtering section

RayOffiah committed:
[DOC-12921]: Feedback on Search Functions | Couchbase Docs (#324)

simon-dew committed:
NO ISSUE: replace callout list for diagram (#330)

rakhi-prathap committed:
[DOC-10857] Add a note to ARRAY_REMOVE() (#334)

rakhi-prathap committed:
[DOC-10023] Add IS UNKNOWN Syntax (#343)

TimLFletcher committed:
DOC-13223 sizing guidelines, note on residentRatio (#347)

* DOC-13223: rebased to 7.6

* DOC-13223: changed to note

* Update index-partitioning.adoc

rakhi-prathap committed:
[DOC-10001] Add new logging qualifiers to system:completed_requests (#361)

rakhi-prathap committed:
[DOC-10909] Add 'Errors' Logging Qualifier (#365)

rakhi-prathap committed:
[DOC 13257] Add new fields to system:completed_requests and system:active_requests (#363)

rakhi-prathap committed:
[DOC-10618] Additional Date Formats (#368)

rakhi-prathap committed:
[DOC-12695] New SQL++ Statement - USING AI (#374)

rakhi-prathap committed:
[DOC-12880] Auto Update Statistics (#360)

RayOffiah committed:
Merge pull request #385

* [DOC-13376]: Feedback on Read Data and Return Results | Couchbase Docs

rakhi-prathap committed:
[DOC-12493] Add XATTR support to SQL++ statements (#388)

ggray-cb committed:
Removing single reference to memcached buckets (#367)

* Removing single reference to memecached bukets

* Removing Couchbase and Ephemeral at Richard's suggestion

Pallavi-Janardhan committed:
DOC-13470: add example code snippet for on deploy (#401)

* Doc_13470_Adding example code snippet for OnDeploy

* DOC_13470_Adding more info from the SME

Changed the info according to SME comments and created a hyperlink to another file.

* Added an xref link to another topic in a diff file

* Added an xref link to the function settings section in a diff file

* Added an xref link to the function settings section in a diff file

* DOC_13470 Adding info on OnDeploy Timeout as per the SME

rakhi-prathap committed:
[DOC-12664] Automatic Workload Repository (AWR) (#342)

simon-dew committed:
DOC-13306: Add documentation for indexer level scan timeout (#399)

* Replace Defer Index Builds example with link
* Replace Indexer Scan Timeout example with link

simon-dew committed:
DOC-13111: cCURL add options cert, key, passphrase for client certificate (#411)

* Preliminary fixes
* Add new points for cacert
* Add cert, key, and passphrase options

rakhi-prathap committed:
[DOC-11953] New options for INFER - 8.0 (#410)

rakhi-prathap committed:
[DOC-13464] Updates to the USING AI statement (#412)

simon-dew committed:
[DOC-9249] Add HASHBYTES Function (#327)

simon-dew committed:
[DOC-11906] Add system:group_info & system:bucket_info keyspaces (#338)

simon-dew committed:
[DOC-13215] Add COMPRESS() and UNCOMPRESS() Functions (#344)

---------

Co-authored-by: rakhi-prathap <rakhi.prathap@couchbase.com>

simon-dew committed:
[DOC-11766] Add EVALUATE Function (#346)

---------

Co-authored-by: rakhi-prathap <rakhi.prathap@couchbase.com>

simon-dew committed:
[DOC-10871] Add OBJECT_CONCAT2 Function (#333)

---------

Co-authored-by: rakhi-prathap <rakhi.prathap@couchbase.com>

simon-dew committed:
[DOC-10792] Add OBJECT_TYPES/OBJECT_TYPES_NESTED functions (#331)

---------

Co-authored-by: rakhi-prathap <rakhi.prathap@couchbase.com>

simon-dew committed:
[DOC-12632] Auto Reprepare for PREPARE Statements (#387)

---------

Co-authored-by: rakhi-prathap <rakhi.prathap@couchbase.com>

simon-dew committed:
DOC-12757 Added information about OnDeploy (#391)

---------

Co-authored-by: Pallavi-Janardhan <pallavi.janardhan@couchbase.com>

simon-dew committed:
[DOC-11152] Changes to date functions (#383)

---------

Co-authored-by: rakhi-prathap <rakhi.prathap@couchbase.com>

simon-dew committed:
[DOC-10551] Additional object parameters and new object_filter function (#376)

---------

Co-authored-by: rakhi-prathap <rakhi.prathap@couchbase.com>

simon-dew committed:
[DOC-10590] Add EXCLUDE clause to SELECT (#382)

---------

Co-authored-by: rakhi-prathap <rakhi.prathap@couchbase.com>

simon-dew committed:
[DOC-10552] Changes to array operators and functions (#371)

---------

Co-authored-by: rakhi-prathap <rakhi.prathap@couchbase.com>

sarahlwelton reviewed (approved):
There's 1 vector search change that snuck in here that's fine to leave - I remember reviewing the PR and I know the changes are appropriate. 

I need to double-check something on one of my PRs that you called out, but some of them were erroneously tagged as 8.0 when they were 7.6 work. I've removed the 8.0 labels from the 7.6 PRs.

simon-dew committed:
DOC-13542: Update Capella REST API builds for Morpheus

- Hide additional description in query settings
- Include credentials in query settings
- Prepare Query AWR for Data API

simon-dew commented:
Merging this with the `AV-91970-8-0-docs-uptake-capella` branch. Further reviews or changes that need to be copied from 8.0 to capella can be added there.
